src/main/java/duke/logging/TaskList.java

```diff
@@ -0,0 +1,155 @@
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+import duke.exception.InvalidDescriptionException;
+import duke.model.Deadline;
+import duke.model.Event;
+import duke.model.Task;
+import duke.model.ToDo;
+
+/**
+ * A TaskList call denotes a system with a list of task and has the ability to make changes to the list.
+ */
+public class TaskList {
+    private final ArrayList<Task> tasks;
+
+    /**
+     * Construct a TaskList.
+     * @param tasks   A list of tasks.
+     */
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Print out all the tasks in the list.
+     */
+    public String list() {
+        String message = "     Here are the tasks in your list:";
+        for (int i = 0; i < tasks.size(); i++) {
+            message += "\n" + "     " + (i + 1) + "." + tasks.get(i);
+        }
+        return message;
+    }
+
+    /**
+     * Add a task into the list of tasks.
+     * @param type                         The type of task.
+     * @param taskDescription              The description of the task.
+     * @param storage                      The storage information containing the list of tasks information.
+     * @return                             The updated list of tasks.
+     * @throws InvalidDescriptionException If description format is wrong.
+     * @throws DateTimeParseException      If datetime format is wrong.
+     * @throws IOException                 If inputting information into the storage post an error.
+     */
+    public ArrayList<Task> addTask(String type, String taskDescription, Storage storage)
+            throws InvalidDescriptionException, DateTimeParseException, IOException {
+        Task task;
+        if (taskDescription.length() == 0) {
+            throw new InvalidDescriptionException("OOPS!!! The description of " + type + " cannot be empty.");
+        } else if (type.equals("todo")) {
+            task = new ToDo(false, taskDescription);
+        } else if ((type.equals("deadline") && !taskDescription.contains("/by"))
+                || (type.equals("event")) && !taskDescription.contains("/at")) {
+            throw new InvalidDescriptionException("OOPS!!! The description format of " + type + " is wrong.");
+        } else {
+            int index = type.equals("deadline")
+                    ? taskDescription.indexOf("/by")
+                    : taskDescription.indexOf("/at");
+            String taskName = taskDescription.substring(0, index);
+            String dateTimeString = taskDescription.substring(index + 4).strip()
+                    .replace("/", "-");
+            LocalDate dateTime = LocalDate.parse(dateTimeString);
+
+            if (type.equals("deadline")) {
+                task = new Deadline(false, taskName, dateTime);
+            } else {
+                task = new Event(false, taskName, dateTime);
+            }
```

Comment:
would it be better to extract out this nested if-else statement,
as a method, so that the reader do not need to trace the code too much?